flow/Stats.h

```diff
@@ -110,6 +121,59 @@ struct SpecialCounter : ICounter, FastAllocated<SpecialCounter<F>> {
 template <class F>
 static void specialCounter(CounterCollection& collection, std::string const& name, F && f) { new SpecialCounter<F>(collection, name, std::move(f)); }
 
+class LatencyBands {
+public:
+	LatencyBands(std::string name, CounterCollection &cc) : name(name), cc(cc), filteredCount(nullptr) {}
+
+	void addThreshold(double value) {
+		if(value > 0 && bands.count(value) == 0) {
+			if(bands.size() == 0) {
+				filteredCount = new Counter(format("Filtered%s", name.c_str()), cc);
+				insertBand(std::numeric_limits<double>::infinity());
+			}
+
+			insertBand(value);
+		}
+	}
+
+	void addMeasurement(double measurement, bool filtered=false) {
+		if(filtered && filteredCount) {
+			++(*filteredCount);
+		}
+		else if(bands.size() > 0) {
+			auto itr = bands.upper_bound(measurement);
+			ASSERT(itr != bands.end());
+			++(*itr->second);
+		}
+	}
+
+	void clearBands() {
+		for(auto itr : bands) {
+			delete itr.second;
+		}
+		
+		bands.clear();
+
+		delete filteredCount;
+	}
+
+	~LatencyBands() {
+		clearBands();
+	}
+
+private:
+	std::map<double, Counter*> bands;
```

Comment:
get rid of Counter object

Reply:
After thinking about this, I think the reason I kept counter objects here is because it makes it easier to write useful queries when you have access to the trace logs. However, in status I think we just need the total count in order to compute the number of these events that occurred over some interval.